charmhelpers/contrib/openstack/context.py

```diff
@@ -955,10 +957,21 @@ def __call__(self):
         db = kv()
         ctxt['stat_password'] = db.get('stat-password')
         if not ctxt['stat_password']:
-            ctxt['stat_password'] = db.set('stat-password',
-                                           pwgen(32))
+            ctxt['stat_password'] = db.set('stat-password', pwgen(32))
             db.flush()
 
+        # NOTE(rgildein): configure prometheus exporter for haproxy > 2.0.0
+        #                 New bind will be created and a prometheus-exporter
+        #                 will be used for path /metrics. At the same time,
+        #                 prometheus-exporter avoids using auth.
+        haproxy_version = get_installed_version("haproxy")
+        if (haproxy_version and
+                haproxy_version.ver_str >= LooseVersion("2.0.0") and
+                is_relation_made("haproxy-exporter")):
+            ctxt["stats_exporter_host"] = get_relation_ip("haproxy-exporter")
+            ctxt["stats_exporter_port"] = \
+                config("haproxy-exporter-stats-port") or 8404
```

Comment:
Just noticed this.  It would be good, here, to verify that the config value is a port and can be turned into an `int`.  e.g. something like

```python
    try:
        ctxt["stats_exporter_port"] = \
            int(config("haproxy-exporter-stats-port")) or 8404
        assert ctxt['stats_exporter_port'] != 0, 'ha-proxy-export-status-port must not be 0'
    except (AssertionError, ValueError, TypeError) as e:
         # do something with the error?  (e.g. raise another exception, or bail out with a log or something else?

```

Reply:
Thanks, I add `haproxy_exporter_stats_port` property, where I convert the port to integer and if an error occurs I return the default value. I'm logging error messages.